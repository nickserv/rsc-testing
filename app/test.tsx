/// <reference types="vitest/globals" />
import { render, screen } from "@testing-library/react";
import { Suspense } from "react";
import Layout from "./layout";
import Page from "./page";

vi.mock("server-only", () => ({}));

test.each(["client", "server", "layout"])("%s component", async (keyword) => {
	render(
		<Suspense>
			<Layout>
				<Page />
			</Layout>
		</Suspense>,
	);

	await screen.findByText(keyword);
});

test("Having <Layout> here will generate an empty snapshot", async () => {
	const { container } = render(
		<Suspense>
			<Layout>
				<Page />
			</Layout>
		</Suspense>,
	);
	await screen.findByText("server");
	expect(container).toMatchSnapshot();
});

test("Don't using screen here will generate an empty snapshot", async () => {
	const { container } = render(
		<Suspense>
			<Layout>
				<Page />
			</Layout>
		</Suspense>,
	);

	expect(container).toMatchSnapshot();
});

test("Removing the <Layout> tag and calling screen generate a snapshot", async () => {
	const { container } = render(
		<Suspense>
			<Page />
		</Suspense>,
	);
	await screen.findByText("server");
	expect(container).toMatchSnapshot();
});
