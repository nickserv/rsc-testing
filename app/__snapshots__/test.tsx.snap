// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Don't using screen here will generate an empty snapshot 1`] = `<div />`;

exports[`Having <Layout> here will generate an empty snapshot 1`] = `<div />`;

exports[`Removing the <Layout> tag and calling screen generate a snapshot 1`] = `
<div>
  <p>
    server
  </p>
  <p>
    client
  </p>
</div>
`;
